mailer.js

```diff
@@ -485,34 +540,72 @@ module.exports = function (log) {
         'X-Verify-Code': message.code
       },
       subject: subject,
-      template: template,
+      template: templateName,
       templateValues: {
+        alternativeLink: alternativeLink,
         email: message.email,
         link: link,
         oneClickLink: oneClickLink,
-        supportUrl: this.supportUrl,
-        supportLinkAttributes: this._supportLinkAttributes()
+        privacyUrl: this.createPrivacyLink(templateName),
+        supportUrl: this.createSupportLink(templateName),
+        supportLinkAttributes: this._supportLinkAttributes(templateName)
       },
       uid: message.uid
     })
   }
 
-  Mailer.prototype.createPasswordResetLink = function (email) {
-    var queryParams = { email: email, reset_password_confirm: false }
+  Mailer.prototype._generateUTMLink = function (link, query, templateName, content) {
+    if (!query) {
+      query = {}
+    }
+
+    query['utm_source'] = 'email'
+    query['utm_medium'] = 'email'
+
+    var campaign = templateNameToCampaignMap[templateName]
+    if (campaign && !query['utm_campaign']) {
+      query['utm_campaign'] = utmPrefix + campaign
+    }
 
-    return this.initiatePasswordResetUrl + '?' + qs.stringify(queryParams)
+    if (content) {
+      query['utm_content'] = utmPrefix + content
+    }
+
+    if (link) {
+      // Check if url contains any query params, if so, append new params and return
+      var parsedLink = url.parse(link)
+      if (parsedLink.query) {
+        return parsedLink.href + '&' + qs.stringify(query)
```

Comment:
Are there any edge cases doing this? If the original query params and the new query params both contain the same key, the result will have duplicate keys. Are you better off merging the two objects and then doing qs.stringify on the merged object?